docs/changes/modeling/16673.feature.rst

```diff
@@ -0,0 +1 @@
+Performance of most non-linear fitters has been significantly improved by reducing the overhead in evaluating models inside the objective function.
```

Comment:
It looks like https://github.com/astropy/astropy/pull/16694 will go ahead, so I think you can pre-emptively rename this to 'perf'